pluginlib/CMakeLists.txt

```diff
@@ -80,7 +80,7 @@ if(BUILD_TESTING)
     )
     target_compile_definitions(${PROJECT_NAME}_unique_ptr_test PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
 
-    if(CMAKE_COMPILER_IS_GNUCXX)
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
```

Comment:
It seems like most of the CMAKE pundits on StackOverflow and mailing lists favor the construct

```cmake
 if (UNIX AND NOT APPLE)
  ...
endif()
```

While it's a bit obtuse I think it's likely what's desired in order to also cover cases like NetBSD, QNX or NuttX.

@wjwwood I would love your advice on whether you think explicitly matching Linux or using this other construct is more appropriate.

Reply:
Not sure if it makes much difference, since it will break one way or another. The only thing I can see that might be better about `if (UNIX AND NOT APPLE)` is that it will break earlier (in pluginlib tests) rather than later (in a package like rviz).

Reply:
> Not sure if it makes much difference, since it will break one way or another.

Not sure I fully understand. What's expected to break? Is it the availability of the c++fs on non-Linux platforms?

Reply:
For example, NuttX may not have `c++fs` (not sure if this is true or not).

Reply:
Okay. That makes sense to me. If the platform doesn't have the required dependency then the build won't succeed one way or the other. But if the dependency is satisfied on a platform the `(UNIX AND NOT APPLE)` query should implement the desired link.

 I'm all in favor of failing earlier and being forward compatible. So let's go with `(UNIX AND NOT APPLE)`.